Wrappers/Python/cil/optimisation/algorithms/Algorithm.py

```diff
@@ -128,20 +128,34 @@ def __next__(self):
             
 
     def update_previous_solution(self):
-        '''Update the previous solution with the current one
+        """ Update the previous solution with the current one
         
         The concrete algorithm calls update_previous_solution. Normally this would 
         entail the swapping of pointers:
 
-        tmp = self.x_old
-        self.x_old = self.x
-        self.x = tmp 
-        '''
+        .. highlight:: python
+        .. code-block:: python
+
+            tmp = self.x_old
+            self.x_old = self.x
+            self.x = tmp 
+        
+
+        """
         pass
         
     def get_output(self):
         '''Returns the solution found'''
         return self.x
+
+    @property
+    def provable_convergence_condition(self):
```

Comment:
This seems wrong to me. I thought we wanted users to call a method `is_provably_convergent` (not a property as the calculation might be costly), which would then call an algorithm specific method that will return True or False.

As it is `_is_provably_convergent` won't ever be called/exposed due to the naming convention.